Add release notes and announcement for 1.14.0 (#131)
anbox-cloud/discourse-docs: announcements/1.14.0.md

## Introduction

The Anbox Cloud team is pleased to announce the release of Anbox Cloud 1.14.0.

Please see [component versions](https://anbox-cloud.io/docs/component-versions) for a list of updated components.

## New features & improvements

> 

Reviewer: 1. Implement syscall interception via LXC seccomp notification support. This enables Anbox to handle different syscalls like setpriority or sched_set_scheduler for processes inside the Android container.
2. Unify the pointer input model by using pointer events to handle mouse and touch events.
3. Switch the memory allocator from the default Glibc based to the tcmalloc to decrease the memory footprint and optimize the memory consumption for Anbox session.

Reviewer 2: Added under "Other" and to the release notes.

### Charmhub migration

> All charms part of Anbox Cloud and used 3rd party charms now coming from [Charmhub](https://charmhub.io) instead of the old Juju Charm Store.

Reviewer 2: ```suggestion
All charms that are part of Anbox Cloud and the used 3rd-party charms now come from [Charmhub](https://charmhub.io) instead of the old Juju charm store.
```

The migration required a rename of the charms as Charmhub only provides a flat namespace.

| Old name | New name |
|----------|----------|
| cs:~anbox-charmers/ams | [ams](https://charmhub.io/ams) |
| cs:~anbox-charmers/ams-node-controller | [ams-node-controller](https://charmhub.io/ams-node-controller) |
| cs:~anbox-charmers/anbox-stream-agent | [anbox-stream-agent](https://charmhub.io/anbox-stream-agent) |
| cs:~anbox-charmers/anbox-stream-gateway | [anbox-stream-gateway](https://charmhub.io/anbox-stream-gateway) |
| cs:~anbox-charmers/lxd | [ams-lxd](https://charmhub.io/ams-lxd) |
| cs:~anbox-charmers/coturn | [coturn](https://charmhub.io/coturn) |
| cs:~anbox-charmers/anbox-cloud-dashboard | [anbox-cloud-dashboard](https://charmhub.io/anbox-cloud-dashboard) |
| cs:~anbox-charmers/ams-load-balancer | [ams-load-balancer](https://charmhub.io/ams-load-balancer) |
| cs:~anbox-charmers/aar | [aar](https://charmhub.io/aar) |

> Please have a look at the [upgrade instructions](https://discourse.ubuntu.com/t/upgrading-from-previous-versions/17750) for further information on how migrate to Charmhub.

Reviewer 2: ```suggestion
See the [upgrade instructions](https://discourse.ubuntu.com/t/upgrading-from-previous-versions/17750) for further information on how migrate to Charmhub.
```

### Anbox images based on Ubuntu 22.04

> Next to the existing Anbox images based on Ubuntu 18.04 Anbox Cloud now also provides Ubuntu 22.04 based ones. From an Anbox functionality perspective there is no difference, the image is just sitting on top of a newer Ubuntu version providing access to newer software from the Ubuntu archive. The Ubuntu 18.04 images are still going to be maintained going forward but will be deprecated at some point in the future (before Ubuntu 18.04 EOL in April 2023).

Reviewer 2: ```suggestion
In addition to the existing Anbox images based on Ubuntu 18.04, Anbox Cloud now provides Ubuntu 22.04 based ones. From an Anbox functionality perspective there is no difference; the image is just sitting on top of a newer Ubuntu version providing access to newer software from the Ubuntu archive. The Ubuntu 18.04 images will still be maintained going forward, but will be deprecated at some point in the future (before Ubuntu 18.04 EOL in April 2023).
```

### Hook support for AMS applications

> [Applications](https://discourse.ubuntu.com/t/manage-applications/24333) in AMS have now support for hooks. Hooks were until now only present as part of [addons](https://discourse.ubuntu.com/t/managing-addons/17759) and are now fully supported with the same feature set by applications.

Reviewer 2: ```suggestion
[Applications](https://discourse.ubuntu.com/t/manage-applications/24333) in AMS now have support for hooks. Until now, hooks were only present as part of [addons](https://discourse.ubuntu.com/t/managing-addons/17759). They are now fully supported by applications, with the same feature set.
```

Adding hooks as part of [applications](https://discourse.ubuntu.com/t/manage-applications/24333) allows in some cases for better structure when functionality does not need to be shared with other applications.

> You can find more information [here](tbd)

Reviewer 2: ```suggestion
You can find more information [here](tbd).
```

### Vulkan support for AMD and Intel GPUs

> For supported AMD and Intel GPUs Vulkan support is now available and allows regular Android applications making use of the Vulkan graphics API to be used.

Reviewer 2: ```suggestion
For supported AMD and Intel GPUs, Vulkan support is now available. This allows using regular Android applications that make use of the Vulkan graphics API.
```

Reviewer 2: Not sure if I resolved that last part correctly?

> ### Multi node support for the Anbox Cloud Appliance (experimental)

Reviewer 2: ```suggestion
### Multi-node support for the Anbox Cloud Appliance (experimental)
```

> The Anbox Cloud Appliance has been meant to be used on a single machine so far but with the 1.14 release experimental support to join the initial machine with additional ones for increased capacity is added. Multi node support does not provide high availability functionality at this time and is purely meant to extend capacity of the underlying LXD cluster.

Reviewer 2: ```suggestion
The Anbox Cloud Appliance was meant to be used on a single machine so far, but with the 1.14 release, experimental support to join the initial machine with additional ones for increased capacity is added. Multi-node support does not provide high availability functionality at this time. It is purely meant to extend the capacity of the underlying LXD cluster.
```

> You can find more information [here](tbd)

Reviewer 2: ```suggestion
You can find more information [here](tbd).
```

### Shiftfs disabled by default

> Until 1.14 Anbox Cloud enabled [shiftfs](https://discuss.linuxcontainers.org/t/trying-out-shiftfs/5155) support in LXD by default. Shiftfs allows for instant creation of containers as no manual shifting of file/directory ownership for extended use of user namespaces is necessary. However, due to bugs which affect the represented file path in `/proc/*/maps` files which some Android applications (and Android itself) run into bugs. The application use the information `/proc/*/maps` to find a path to certain libraries or memory segments they need to access. As shiftfs is being deprecated and the replacement ([idmapped mounts](https://lore.kernel.org/lkml/20210213130042.828076-1-christian.brauner@ubuntu.com/T/#u)) is not ready for the ZFS filesystem Anbox Cloud will turn off shiftfs by default starting with 1.14.0.

Reviewer 2: ```suggestion
Until 1.14, Anbox Cloud enabled [shiftfs](https://discuss.linuxcontainers.org/t/trying-out-shiftfs/5155) support in LXD by default. Shiftfs allows for instant creation of containers, because no manual shifting of file/directory ownership for extended use of user namespaces is necessary. However, there are bugs that affect the represented file path in `/proc/*/maps` files, which some Android applications (and Android itself) use. The applications use the information in `/proc/*/maps` to find a path to certain libraries or memory segments they need to access. 

As shiftfs is being deprecated and the replacement ([idmapped mounts](https://lore.kernel.org/lkml/20210213130042.828076-1-christian.brauner@ubuntu.com/T/#u)) is not ready for the ZFS file system, Anbox Cloud turns off shiftfs by default starting with 1.14.0.
```

Reviewer 2: Can you check the text again? I didn't understand the third sentence (I think it's missing some parts).

> If you want to turn it back on, you can simple set the configuration option on the LXD charm

Reviewer 2: ```suggestion
If you want to turn it back on, set the following configuration option on the LXD charm:
```

    juju config lxd shiftfs_enabled=true

### Other

 * The NVIDIA 510.x driver series is now the default
 * Android security updates for May 2022 (see [Android Security Bulletin - May 2022](https://source.android.com/security/bulletin/2022-05-01) for more information)
 * Android WebView has been updated to [101.0.4951.41](https://chromereleases.googleblog.com/2022/04/stable-channel-update-for-desktop_26.html)

## Known issues

> n/a

Reviewer 2: Maybe this should be added to the release roadmap? https://anbox-cloud.io/docs/roadmap
It isn't really related to 1.14, so not sure it should go into the announcement.

> n/a

Reviewer: That looks good to me too.

> n/a

Reviewer 2: Created https://warthogs.atlassian.net/browse/AC-899

## Bug fixes

> * AC-877 Fix the bugs of JS SDK after the recent refactor
> * AC-867 Android 12 image ships com.android.emulator.multidisplay
> * AC-836 vkcube crashes without support for HAL\_PIXEL\_FORMAT\_RGBA\_1010102
> * AC-835 Memory leaks happened to Anbox session process after a container running for a long time
> * AC-833 anbox host not set in /etc/hosts in Android 12
> * AC-832 Android CTS tests report /proc isn't mounted with hidepid=2
> * AC-821 CTS test cases for system features fail
> * AC-814 AMS dashboard doesn't load on the Appliance
> * AC-810 Failing gateway docs build fails and doesn't stop CI from succeeding
> * AC-741 End touch/mouse events when leaving the SDK container
> * AC-699 dEQP-GLES3.functional.texture.shadow.cube.linear.greater\_or\_equal\_depth24\_stencil8 fails on swrast
> * AC-693 SDK is broken on iOS on 4G only
> * AC-548 Colors with direct software rendering in scrcpy are incorrect

Reviewer 2: ```suggestion
* AC-877 Fix the bugs of JS SDK after the recent refactor
* AC-867 Android 12 image ships `com.android.emulator.multidisplay`
* AC-836 `vkcube` crashes without support for `HAL_PIXEL_FORMAT_RGBA_1010102`
* AC-835 Memory leaks happened to Anbox session process after a container running for a long time
* AC-833 anbox host not set in `/etc/hosts` in Android 12
* AC-832 Android CTS tests report `/proc` isn't mounted with `hidepid=2`
* AC-821 CTS test cases for system features fail
* AC-814 AMS dashboard doesn't load on the Appliance
* AC-810 Failing gateway docs build fails and doesn't stop CI from succeeding
* AC-741 End touch/mouse events when leaving the SDK container
* AC-699 `dEQP-GLES3.functional.texture.shadow.cube.linear.greater_or_equal_depth24_stencil8` fails on `swrast`
* AC-693 SDK is broken on iOS on 4G only
* AC-548 Colours with direct software rendering in scrcpy are incorrect
```

[Applications](https://discourse.ubuntu.com/t/manage-applications/24333) in AMS now have support for hooks. Until now, hooks were only present as part of [addons](https://discourse.ubuntu.com/t/managing-addons/17759). They are now fully supported by applications, with the same feature set.

Adding hooks as part of [applications](https://discourse.ubuntu.com/t/manage-applications/24333) allows in some cases for better structure when functionality does not need to be shared with other applications.

You can find more information [here](tbd).

### Vulkan support for AMD and Intel GPUs

For supported AMD and Intel GPUs, Vulkan support is now available. This allows using regular Android applications that make use of the Vulkan graphics API.

### Multi-node support for the Anbox Cloud Appliance (experimental)

The Anbox Cloud Appliance was meant to be used on a single machine so far, but with the 1.14 release, experimental support to join the initial machine with additional ones for increased capacity is added. Multi-node support does not provide high availability functionality at this time. It is purely meant to extend the capacity of the underlying LXD cluster.

> You can find more information [here](tbd).

Reviewer 2: Removing this link for now, since we don't have documentation for this yet.
